### ISSUE TITLE
gpg-agent from gnupg21 crashes

### ISSUE THREAD

**Issue by aabdnn** [Homebrew/homebrew-versions#1459]
### Bug reports:

On MacOS Sierra, with a fully up to date homebrew/versions tap and no other problems, gpg-agent from the gnupg21 formula crashes.

```
$ gpg-agent 
Assertion failed: (res == 0), function enter_npth, file npth.c, line 123.
Abort trap: 6
```

**Comment by ilovezfs** [comment]
Yup, thanks for reporting this.

**Comment by ilovezfs** [comment]
happens on 10.11 too.

**Comment by ilovezfs** [comment]
This is a regression due to the upgrade to libgpg-error 1.25 https://github.com/Homebrew/homebrew-core/pull/6989. Unfortunately rebuilding doesn't fix it so we'll need to investigate further whether that upgrade has to be rolled back or patched.

**Comment by RomainMuller** [comment]
I'm having the same issue. As @aabdnn noted on https://github.com/Homebrew/homebrew-core/pull/6989, there is no easy way to downgrade `libgpg-error` and recover a working `gnupg21`. Could it be rolled back while the incompatibility is being figured out?

**Comment by woodruffw** [comment]
I merged 1.25, so I'll roll it back. Apologies to all for the broken upgrade.

**Comment by ilovezfs** [comment]
@woodruffw we may want to just wait as it's already fixed upstream in gnupg21 head and the release is due out imminently.

**Comment by RomainMuller** [comment]
Makes sense to me.

**Comment by ilovezfs** [comment]
@aabdnn @RomainMuller mind trying https://github.com/Homebrew/homebrew-versions/pull/1460?

**Comment by RomainMuller** [comment]
@ilovezfs - it works for me. Thanks for te quick turnaround!

**Comment by aabdnn** [comment]
@ilovezfs thanks for this. I only just saw it. In the meantime though, I managed to downgrade libgpg-error to 1.24 and pinned it, because I really needed gpg to work urgently. It's past midnight here for me, and I'm exhausted, so if you still need me to, I can try this patch tomorrow morning.

**Comment by grempe** [comment]
Ugh. Glad I found this issue. Hours wasted installing/uninstalling/building from source & fruitless googling.

If there are instructions to pin to libgpg-error 1.2.4 that would be helpful while we wait for the jopefully imminent critical patch release (or the new gpg21 version release).

**Comment by ilovezfs** [comment]
PR for the new version: https://github.com/Homebrew/homebrew-versions/pull/1461

**Comment by ilovezfs** [comment]
This should be fixed now if you brew update and upgrade to 2.1.16
